### PR TITLE
mole: init at 2.0.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
CLI application to create SSH tunnels

https://github.com/davrodpin/mole

Related to #81418

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 140581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mole</li>
  </ul>
</details>

File: pkgs/tools/networking/mole/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "davrodpin";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "11q48wfsr35xf2gmvh4biq4hlpba3fh6lrm3p9wni0rl1nxy40i7";
+  };
+
+  vendorSha256 = "1qm328ldkaifj1vsrz025vsa2wqzii9rky00b6wh8jf31f4ljbzv";
+
+  #doCheck = false;
+
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/mole/default.nix:20]:
```suggestion
```

Comment (fabaff) [reply pkgs/tools/networking/mole/default.nix]: Thanks, removed.

File: pkgs/tools/networking/mole/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "davrodpin";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "11q48wfsr35xf2gmvh4biq4hlpba3fh6lrm3p9wni0rl1nxy40i7";
+  };
+
+  vendorSha256 = "1qm328ldkaifj1vsrz025vsa2wqzii9rky00b6wh8jf31f4ljbzv";
```
Comment (figsoda) [inline pkgs/tools/networking/mole/default.nix:17]:
```ShellSession
[nix-shell:~/.cache/nixpkgs-review/pr-140581]$ mole version
unversioned
```
```suggestion
  vendorSha256 = "1qm328ldkaifj1vsrz025vsa2wqzii9rky00b6wh8jf31f4ljbzv";
  
  ldflags = [ "-X=github.com/davrodpin/mole/cmd.version=${version}" ];
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/mole/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/tools/networking/mole/default.nix:17]: Thanks, added.

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 140581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mole</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**